parquet-column/src/test/java/org/apache/parquet/column/values/bloom/Murmur3.java

```diff
@@ -0,0 +1,333 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.parquet.column.values.bloom;
+
+/**
+ * Murmur3 is successor to Murmur2 fast non-crytographic hash algorithms.
+ *
+ * Murmur3 32 and 128 bit variants.
+ * 32-bit Java port of https://code.google.com/p/smhasher/source/browse/trunk/MurmurHash3.cpp#94
+ * 128-bit Java port of https://code.google.com/p/smhasher/source/browse/trunk/MurmurHash3.cpp#255
+ *
+ * This is a public domain code with no copyrights.
```

Comment:
Why not use the code directly and use JNI to call it?

Reply:
it will introduce a c++ dependency, not sure whether current parquet-mr has any C++ dependency yet. Also I didn't see any formal release of Murmur3.

Reply:
What methodology have you used to check that your Java translation computes the same values as the C++ version?

Reply:
I checked it manually.

Reply:
There are implementations of Murmur3 available elsewhere. Google guava has one, for example, that handles ints and longs directly. I don't think we should include this.